dhis-2/dhis-services/dhis-service-administration/src/main/java/org/hisp/dhis/maintenance/DefaultMaintenanceService.java

```diff
@@ -146,6 +146,16 @@ public int deleteSoftDeletedProgramStageInstances()
         return result;
     }
 
+    @Override
+    public int deleteSoftDeletedRelationships()
+    {
+        int result = maintenanceStore.deleteSoftDeletedRelationships();
+
+        log.info( "Permanently deleted soft deleted relationships: " + result );
```

Comment:
I think such a log statement is more for devs. So, I would see this as a debug log. I would not add this line. I think we log too much. As an operator of DHIS2 I don't have to act on this. All is fine, so I would not need to read this line.

Use parametrized messages to not construct the string if the log level is disabled.
https://www.slf4j.org/faq.html#logging_performance

You also have an audit statement later on, so this information is available in the audit already.

Reply:
This is a once in a blue moon action, and we also log for all similar operations. I think logging is fine, but as @teleivo  says, info might not be the right level. But thats something that should be changed for all the logs in that case.

To summarize, I think it's fine to add it with info as it is now, since thats what the routine has been for this kind of action. If we do decide to increase log level, we should create a new ticket.